app/js/ContentCacheManager.js

```diff
@@ -219,6 +233,28 @@ async function writePdfStream(dir, hash, buffer) {
   }
 }
 
+function getDeadlineChecker(compileTime) {
+  const maxOverhead = Math.min(
+    // Adding 10s to a 40s compile time is OK.
```

Comment:
let's also allow anything up to 1000ms regardless.. if it only takes a second on a fast compile that's ok as far as limiting the damage.

Reply:
Done in https://github.com/overleaf/clsi/pull/254/commits/756d4a14a18254e217330c606e2a74416af5ff3d